presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -315,17 +317,22 @@ public FileStatus getFileStatus(Path path)
         }
 
         return new FileStatus(
-                getObjectSize(metadata),
+                getObjectSize(path, metadata),
                 false,
                 1,
                 BLOCK_SIZE.toBytes(),
                 lastModifiedTime(metadata),
                 qualifiedPath(path));
     }
 
-    private static long getObjectSize(ObjectMetadata metadata)
+    private static long getObjectSize(Path path, ObjectMetadata metadata)
     {
-        String length = metadata.getUserMetadata().get(UNENCRYPTED_CONTENT_LENGTH);
+        Map<String, String> userMetadata = metadata.getUserMetadata();
+        String length = userMetadata.get(UNENCRYPTED_CONTENT_LENGTH);
+        String sse = userMetadata.get(SERVER_SIDE_ENCRYPTION);
+        if (sse != null && length == null) {
+            throw new RuntimeException(format("%s header is not set on an encrypted object: %s", UNENCRYPTED_CONTENT_LENGTH, path));
```

Comment:
Use `IOException` so that the error gets categorized properly (we don't expect FileSystem to throw runtime exceptions so they leak out)